test/native/setup.js

```diff
@@ -110,22 +109,10 @@ jest.mock( '@react-native-community/blur', () => () => 'BlurView', {
 // https://github.com/facebook/react-native/blob/HEAD/jest/setup.js
 // to fix issue "TypeError: Cannot read property 'Commands' of undefined"
 // raised when calling focus or blur on a native component
-const mockNativeModules = {
-	UIManager: {
-		...NativeModules.UIManager,
-		getViewManagerConfig: jest.fn( () => ( { Commands: {} } ) ),
-	},
-};
-
-Object.keys( mockNativeModules ).forEach( ( module ) => {
-	try {
-		jest.doMock( module, () => mockNativeModules[ module ] ); // needed by FacebookSDK-test
-	} catch ( error ) {
-		jest.doMock( module, () => mockNativeModules[ module ], {
-			virtual: true,
-		} );
-	}
-} );
+const UIManager = jest.requireMock(
+	'react-native/Libraries/ReactNative/UIManager'
```

Comment:
Rather than continuing the pattern of referencing internal module paths, what do you think about using [the approach outlined here](https://github.com/facebook/react-native/issues/26579#issuecomment-538610849) and expanded upon further [in this blog post](https://altany.github.io/react-native/0.61/jest/mocking/upgrade/2020/01/25/mocking-react-native-0.61-modules-with-jest.html)?

I don't necessarily have a strong opinion but, as mentioned in the thread/post, referencing the internal module paths can be brittle.

Reply:
Yeah, I think that was the way this part was originally designed but as far as I checked it wasn't working properly. I'll review the documents you passed and update the approach, thanks!

Reply:
I could be wrong, but I think the existing implementation likely was not working because it is attempting to mock `UIManager` [by name](https://github.com/WordPress/gutenberg/blob/c993a862483510e8fab9e949201a8ce89cb7c54a/test/native/setup.js#L122), which is not supported in `react-native@>0.61`, rather than by path. The alternative I outlined _should_ work because it involves mocking via the top-level `react-native` module name, rather than by the name of a named export.

All-in-all, I think moving forward with the internal path is fine if the alternative proves difficult or brittle. Just let me know! 😄

Reply:
>I could be wrong, but I think the existing implementation likely was not working because it is attempting to mock UIManager by name, which is not supported in react-native@>0.61, rather than by path. The alternative I outlined should work because it involves mocking via the top-level react-native module name, rather than by the name of a named export.

I think you're right, probably that was implemented way before RN 0.61 and since it wasn't being used no one spotted the issue. The alternative you outlined actually is more aligned to the original solution so I'm going to stick with it.

>All-in-all, I think moving forward with the internal path is fine if the alternative proves difficult or brittle. Just let me know! 😄

I don't think it's going to be difficult to change it but I'll let you know if I find any blocker, thanks for the help!

Reply:
I've tried the outlined solution but I'm afraid it's not working for me, as far as I checked it it would require to mock every module you need from `react-native` which could be overkilling.

If you have some spare time I'd appreciate if you could take a look just in case I'm missing something.

I added the following lines to the [setup.js file](https://github.com/WordPress/gutenberg/blob/rnmobile/fix/jest-mock-native-uimanager/test/native/setup.js) and run `npm run native test` in Gutenberg project (if you're on Gutenberg mobile you can run `npm run core test`):
```
import * as ReactNative from 'react-native';

jest.doMock( 'react-native', () => {
	return Object.setPrototypeOf(
		{
			UIManager: {
				...ReactNative.UIManager,
				getViewManagerConfig: jest.fn(),
			},
		},
		ReactNative
	);
} );
```

Reply:
@fluiddot I spent some time exploring the state of the mocks. I found some interesting things that I have captured in #29390. I'd recommend reviewing each commit individually to see what I found in the commit messages.

You are welcome to keep my commits or discard them. Let me know if you'd like to chat about any of them.

Reply:
Thank you very much @dcalhoun for the exploration, awesome work!

I'll review the commits and if I have any comments I'll do it in that [PR](https://github.com/WordPress/gutenberg/pull/29390). As far as I checked, they are good improvements so if possible I'd like to include them in this branch.